src/JWT/JwtValidator.cs

```diff
@@ -32,19 +32,25 @@ public void Validate(string payloadJson, string decodedCrypto, string decodedSig
 
             var now = _dateTimeProvider.GetNow();
             var secondsSinceEpoch = Math.Round((now - UnixEpoch).TotalSeconds);
+            const string claimMustBeDoubleFormat = "Claim '{0}' must be a double.";
 
             // verify exp claim https://tools.ietf.org/html/draft-ietf-oauth-json-web-token-32#section-4.1.4
             object expObj;
             if (payloadData.TryGetValue("exp", out expObj))
             {
-                int expInt;
+                if (expObj == null)
+                {
+                    throw new SignatureVerificationException(string.Format(claimMustBeDoubleFormat, "exp"));
+                }
+
+                double expInt;
                 try
                 {
-                    expInt = Convert.ToInt32(expObj);
+                    expInt = Convert.ToDouble(expObj);
                 }
                 catch
                 {
-                    throw new SignatureVerificationException("Claim 'exp' must be an integer.");
+                    throw new SignatureVerificationException(string.Format(claimMustBeDoubleFormat, "exp"));

```

Comment:
Let's don't extract strings into const variables to keep the source code smaller and more readable